components/cam/src/physics/cam/micro_p3_interface.F90

```diff
@@ -362,6 +367,8 @@ subroutine micro_p3_init(pbuf2d)
        call pbuf_set_field(pbuf2d, relvar_idx, 2._rtype)
        call pbuf_set_field(pbuf2d, accre_enhan_idx, micro_mg_accre_enhan_fac)
        call pbuf_set_field(pbuf2d, qr_evap_tend_idx,  0._rtype)
+       call pbuf_set_field(pbuf2d, qv_prev_idx,  0._rtype)
```

Comment:
Do we really want to set qv_prev and t_prev to 0.0 at intialization?  It seems to me that would imply a pretty big jump in both values for evaporate_sublim.  Would it be better to initialize to the initial value of qv and t?

Reply:
Good point. As long as we don't initialize model simulations with rain water, I think it shouldn't produce a shock (since I think qR = 0), but I agree that it would be better to initialize with values of qv and t. 
Is there a clean way of doing this?

Reply:
hmmm, yes, it could be a bit tricky.  p3_init doesn't really know about the values of qv and t, so we couldn't do it there.  Instead we need to put some kind of check in p3_tend in the micro_p3_interface.F90.  

One possibility is just an "is_first_step" check, and if so set qv_prev = qv and t_prev = t.  However, P3 is substepped, so the first mac_mic_num_steps of p3 would all have is_first_step=.true.

Alternatively we could use pbuf_set_field to set the qv_prev and t_prev to -999.0 (or something equally extreme) and then have 
if (is_first_step) then
  if (qv_prev(1,1) == -999) then
    qv_prev(:,:) = qv(:,:)
    t_prev(:,:) = t(:,:)
  end if
end if

Reply:
I did a quick test to see if changing the initialization value to 999 would lead to BFB answers (following my guess above), and found that it changes the simulation, so I like your second idea. If we set the first qv_prev and t_prev to an odd number (-12), we could even get rid of the is_first_step condition. I can put in a commit that makes this change.

Reply:
I added a commit where I've initialized qv_prev and t_prev to -12 and added a check within evap_precip to set t_prev & qv_prev to t & qv if the value is less than -10. I've confirmed that this produces BFB results even if I change the initialized value to something like -900. Feel free to change it for a cleaner execution.